client/src/JobPaginator.jsx

```diff
@@ -1,326 +1,64 @@
-                            " for this job before publishing it.";
-                        alert(msg);
-                        return;
-                    }
-                    msg =
-                        "Are you sure you wish to publish this job? You will not be able" +
-                        " to make any further edits.";
-                    if (!window.confirm(msg)) {
-                        return;
-                    }
-                }
-                this.ajax.ajax({
-                    type: "PUT",
-                    url: this.ajax.getJobUrl(this.state.category, this.state.job, ""),
-                    contentType: "application/json",
-                    data: JSON.stringify({ order: this.state.order, published: publish }),
-                    success: function () {
-                        alert("Success!");
-                        window.location.assign("/paginate"); // TODO: Route better?
-                        dispatchEvent(new Event("Prep.saved"));
-                    },
-                    error: function () {
-                        alert("Unable to save!");
-                    },
-                });
-            }.bind(this)
-        );
-    }
-
-    toggleZoom() {
-        this.newState = this.state;
-        this.newState.zoom = !this.newState.zoom;
-        this.setState(this.newState);
-    }
-
-    componentDidMount() {
-        this.loadJob(this.props.category, this.props.job);
-    }
-
-    render() {
-        var preview = this.state.zoom ? (
-            <PaginatorZoomy img={this.getImageUrl(this.state.currentPage, "large")} />
-        ) : (
-            <PaginatorPreview img={this.getImageUrl(this.state.currentPage, "medium")} />
-        );
-        return (
-            <div className={this.state.active ? "" : "hidden"} id="paginator">
-                <div className="row">
-                    <div className="six col">{preview}</div>
-                    <div className="six col">
-                        <p>
-                            {this.state.category} &gt; {this.state.job}
-                        </p>
-                        <PaginatorControls paginator={this} />
-                        <PaginatorList paginator={this} pageCount={this.state.order.length} />
-                    </div>
+import { useJobPaginator } from "./JobPaginatorState";
+
+const JobPaginator = ({ initialCategory, initialJob }) => {
+    const {
+        state: { active, currentPage, zoom, order, category, job },
+        action: {
+            setLabel,
+            getLabel,
+            getMagicLabel,
+            setPage,
+            nextPage,
+            prevPage,
+            deletePage,
+            save,
+            autonumberFollowingPages,
+            getImageUrl,
+            setZoom,
+        },
+    } = useJobPaginator(initialCategory, initialJob);
+
+    return (
+        <div className={active ? "" : "hidden"} id="paginator">
```

Comment:
I suspect that "active" can be totally eliminated here, since (as with one of the other components you worked on) it was only here to hide the component in the days before we introduced routing to the application.

Reply:
(I do see that this is apparently used to hide everything while data is loading, so maybe that's useful... but I'd be interested to see if there's a negative consequence to disabling those calls).

Reply:
let me test it now.

Reply:
made change. I'm not sure if it's okay or not.

Reply:
There is a brief flash as the data loads, where the control switches from an empty state to a populated state... but I think that's okay for now. In the longer-term, I think a better solution is to make this control behave more correctly when the order is empty. There are some jobs which do not have any pages (e.g. audio jobs, PDF-only jobs), and it would be nice to have a more sensible display in those cases. I think it may make sense to pass the order.length into the PaginatorControls and hide all of the page-specific controls when there are no pages. Similarly, it may make sense for the PaginatorList to display a message like "No pages found." in place of the thumbnail list when pageCount is empty. This is not urgent, though -- maybe the best thing for now is to create a card for it and deal with it later.

Reply:
https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.planner/tt.c_19:54c0a65e7f5f491aa6f90021330eb74e@thread.tacv2_p_LH9rYXUCYke47Igm88R2rGQAGFab_h_1612357088580?tenantId=765a8de5-cf94-44f0-9caf-ae5bf8cfa366&webUrl=https%3A%2F%2Ftasks.teams.microsoft.com%2Fteamsui%2FpersonalApp%2Falltasklists&context=%7B%22subEntityId%22%3A%22%2Fboard%2Ftask%2F-fOapF7EMEyz0VaK771xXmQAI2KZ%22%2C%22channelId%22%3A%2219%3A54c0a65e7f5f491aa6f90021330eb74e%40thread.tacv2%22%7D

Reply:
https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.planner/tt.c_19:54c0a65e7f5f491aa6f90021330eb74e@thread.tacv2_p_LH9rYXUCYke47Igm88R2rGQAGFab_h_1612357088580?tenantId=765a8de5-cf94-44f0-9caf-ae5bf8cfa366&webUrl=https%3A%2F%2Ftasks.teams.microsoft.com%2Fteamsui%2FpersonalApp%2Falltasklists&context=%7B%22subEntityId%22%3A%22%2Fboard%2Ftask%2F-fOapF7EMEyz0VaK771xXmQAI2KZ%22%2C%22channelId%22%3A%2219%3A54c0a65e7f5f491aa6f90021330eb74e%40thread.tacv2%22%7D

Reply:
Created these two tasks.